### ISSUE TITLE
Separate NSURLDownloadSession instances is inefficient

### ISSUE THREAD

**Issue by hofman** [sparkle-project/Sparkle#1380]
Using separate `NSURLDownloadSession` instances is very inefficient. It is better to use a shared session. This is in particular true for downloading the appcast, as that happens often, while downloading a new app will usually happen never or only once. However, the temporary downloads for the appcast don't use (most of) the delegate methods, so they could easily be handled by the `sharedSession` and a callback based data task.

**Comment by kornelski** [comment]
We want to set QoS (since updates should be low priority, and not interfering with anything else), so we can't use `sharedSession`.

By default the check is done only once a day, so I think it's fine to create a session specifically for it, and not keep it around.
